lua/lang/scala.lua

```diff
@@ -0,0 +1,80 @@
+local M = {}
+
+M.config = function()
+  O.lang.scala = {
+    metals = {
+      active = false,
+      server_version = "0.10.5",
+      excluded_packages = {},
+      show_implicit_arguments = false,
+      show_inferred_type = true,
+      status_bar_provider = false,
+    },
+    formatter = {
+      exe = "scalafmt",
```

Comment:
So a bit unsure on what this is actually doing, but if it's doing what I think it is and actually executing `scalafmt` directly I'd recommend not doing this. Metals has formatting build it using Scalafmt, so all the user needs to do is send in a `vim.lsp.buf.formating()` request while Metals is attached and it will format your file.

Reply:
this is user-configurable as well
user can enable auto formatting by settings this in their config
```lua
O.format_on_save = true
```
also user can manually format the code using:
```
<leader>lf
```

Reply:
But is this executing `scalafmt` directly? It will be slower this way then just relying on the built-in way to do this via lsp.

Reply:
Yes, I agree with @ckipp01 here. `metals` does formatting automatically using scalafmt. We do not need to call it explicitly here.

Reply:
let me remove it then :running_man:

Reply:
@ChristianChiarulli what do you think about this?
I can override the `<leader>lf` or `autoformat augroup` for scala so  instead of running `scalafmt` it just runs
`vim.lsp.buf.formatting()` but sure if it is a good idea or not:

Reply:
as described in #996 , we want to have the ability to use builtin formatter for other langs as well.
so we need a bit of refactoring. but for the moment we gonna let this stay as is

@kpritam @ckipp01 thank you for bringing up this issue